### ISSUE TITLE
Compatibility with latest 2024.3 IntelliJ

### ISSUE THREAD

**Issue by vvandens** [chengpohi/elasticshell#48]
First of all, thank you very much for the plugin. It's of great help using OpenSearch !

**Describe the bug**
Latest IntelliJ version 2024.3 is out and there seems to be several problems with the EDQL plugin on this release. Do you observe the same ? (IntelliJ ultimate 2024.3) :

1. Kepps poping up : The plugin com.github.chengpohi failed to save settings. Please restart IntelliJ IDEA
2. Maybe you are using some private IntelliJ classes that have been removed :

```
java.lang.NoClassDefFoundError: com/intellij/json/JsonBundle
	at com.github.chengpohi.edql.formatter.EDQLCodeStyleSettings$PropertyAlignment.getDescription(EDQLCodeStyleSettings.java:56)
	at com.github.chengpohi.edql.formatter.EDQLLanguageCodeStyleSettingsProvider$Holder.lambda$static$0(EDQLLanguageCodeStyleSettingsProvider.java:22)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at com.github.chengpohi.edql.formatter.EDQLLanguageCodeStyleSettingsProvider$Holder.<clinit>(EDQLLanguageCodeStyleSettingsProvider.java:23)
	at com.github.chengpohi.edql.formatter.EDQLLanguageCodeStyleSettingsProvider.customizeSettings(EDQLLanguageCodeStyleSettingsProvider.java:81)
	at com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider$SupportedFieldCollector.collectFields(LanguageCodeStyleSettingsProvider.java:270)
	at com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider.getSupportedFields(LanguageCodeStyleSettingsProvider.java:256)
	at com.intellij.psi.codeStyle.CommonCodeStyleSettings.writeExternal(CommonCodeStyleSettings.java:183)
	at com.intellij.psi.codeStyle.CommonCodeStyleSettingsManager.writeCommonSettings(CommonCodeStyleSettingsManager.java:237)
	at com.intellij.psi.codeStyle.CommonCodeStyleSettingsManager.writeExternal(CommonCodeStyleSettingsManager.java:217)
	at com.intellij.psi.codeStyle.CodeStyleSettings.writeExternal(CodeStyleSettings.java:629)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleSchemeImpl.writeScheme(CodeStyleSchemeImpl.java:133)
	at com.intellij.configurationStore.LazySchemeProcessor.writeScheme(scheme-impl.kt:70)
	at com.intellij.configurationStore.LazySchemeProcessor.writeScheme(scheme-impl.kt:60)
	at com.intellij.configurationStore.schemeManager.SchemeDataHolderImpl.updateDigest(SchemeDataHolderImpl.kt:33)
	at com.intellij.configurationStore.schemeManager.SchemeDataHolderImpl.updateDigest(SchemeDataHolderImpl.kt:14)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleSchemeImpl.getCodeStyleSettings(CodeStyleSchemeImpl.java:89)
	at com.intellij.psi.codeStyle.CodeStyleSettingsManager.getCurrentSettings(CodeStyleSettingsManager.java:299)
	at com.intellij.application.options.CodeStyle.getDefaultSettings(CodeStyle.java:50)
	at com.intellij.application.options.CodeStyle.getSettings(CodeStyle.java:186)
	at com.intellij.openapi.editor.impl.EditorSettingsState.myWrapWhenTypingReachesRightMargin_delegate$lambda$4(EditorSettingsState.kt:103)
	at com.intellij.openapi.editor.impl.EditorSettingsState.<init>(EditorSettingsState.kt:400)
	at com.intellij.openapi.editor.impl.SettingsImpl.<init>(SettingsImpl.kt:89)
	at com.intellij.openapi.editor.impl.EditorImpl.<init>(EditorImpl.java:381)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.doCreateEditor(EditorFactoryImpl.kt:211)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.kt:179)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.kt:144)
	at com.intellij.ui.EditorTextField.createEditor(EditorTextField.java:695)
	at com.intellij.ui.LanguageTextField.createEditor(LanguageTextField.java:116)
	at com.intellij.ui.EditorTextFieldProviderImpl$MyEditorTextField.createEditor(EditorTextFieldProviderImpl.java:34)
	at com.intellij.ui.EditorTextField.initEditorInner(EditorTextField.java:559)
	at com.intellij.ui.EditorTextField.initEditor(EditorTextField.java:544)
	at com.intellij.ui.EditorTextField.getEditor(EditorTextField.java:994)
	at com.intellij.vcs.commit.NonModalCommitPanel.getButtonPanelBackground(NonModalCommitPanel.kt:139)
	at com.intellij.vcs.commit.NonModalCommitPanel.<init>(NonModalCommitPanel.kt:84)
	at com.intellij.vcs.commit.NonModalCommitPanel.<init>(NonModalCommitPanel.kt:39)
	at com.intellij.vcs.commit.ChangesViewCommitPanel.<init>(ChangesViewCommitPanel.kt:42)
	at com.intellij.openapi.vcs.changes.ChangesViewWorkflowManager.updateCommitWorkflowHandler(ChangesViewWorkflowManager.java:66)
	at com.intellij.openapi.vcs.changes.ChangesViewWorkflowManager.lambda$new$0(ChangesViewWorkflowManager.java:42)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:765)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:712)
	at com.intellij.util.messages.impl.MessageBusImplKt.executeOrAddToQueue(MessageBusImpl.kt:533)
	at com.intellij.util.messages.impl.MessagePublisher.publish$intellij_platform_core(MessageBusImpl.kt:504)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:481)
	at jdk.proxy14/jdk.proxy14.$Proxy160.commitModeChanged(Unknown Source)
	at com.intellij.vcs.commit.CommitModeManager.updateCommitMode(CommitModeManager.kt:82)
	at com.intellij.vcs.commit.CommitModeManager.access$updateCommitMode(CommitModeManager.kt:45)
	at com.intellij.vcs.commit.CommitModeManager$MyStartupActivity$execute$2.invokeSuspend$lambda$0(CommitModeManager.kt:62)
	at com.intellij.openapi.application.CoroutinesKt.writeIntentReadAction$lambda$1$lambda$0(coroutines.kt:329)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction(AnyThreadWriteThreadingSupport.kt:128)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteIntentReadAction(ApplicationImpl.java:916)
	at com.intellij.openapi.application.CoroutinesKt.writeIntentReadAction$lambda$1(coroutines.kt:329)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContextInner(coroutines.kt:341)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invokeSuspend(coroutines.kt:233)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:62)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:261)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:232)
	at com.intellij.openapi.application.CoroutinesKt.writeIntentReadAction(coroutines.kt:328)
	at com.intellij.vcs.commit.CommitModeManager$MyStartupActivity$execute$2.invokeSuspend(CommitModeManager.kt:60)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at com.intellij.openapi.application.impl.DispatchedRunnable.run(DispatchedRunnable.kt:44)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:236)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:25)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:218)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:117)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:728)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:750)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:675)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:573)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18$lambda$17$lambda$16$lambda$15(IdeEventQueue.kt:355)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:857)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18$lambda$17$lambda$16(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2$lambda$1(IdeEventQueue.kt:1045)
	at com.intellij.openapi.application.WriteIntentReadAction.lambda$run$0(WriteIntentReadAction.java:24)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction(AnyThreadWriteThreadingSupport.kt:128)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteIntentReadAction(ApplicationImpl.java:916)
	at com.intellij.openapi.application.WriteIntentReadAction.compute(WriteIntentReadAction.java:55)
	at com.intellij.openapi.application.WriteIntentReadAction.run(WriteIntentReadAction.java:23)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2(IdeEventQueue.kt:1045)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$3(IdeEventQueue.kt:1054)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:109)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1054)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:395)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.ClassNotFoundException: com.intellij.json.JsonBundle PluginClassLoader(plugin=PluginDescriptor(name=Elasticsearch Query - EDQL, id=com.github.chengpohi, descriptorPath=plugin.xml, path=~/Library/Application Support/JetBrains/IntelliJIdea2024.3/plugins/edql-plugin, version=1.10.15, package=null, isBundled=false), packagePrefix=null, state=active, parents=PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.impl, descriptorPath=intellij.platform.vcs.impl.xml, path=~/Applications/IntelliJ IDEA Ultimate.app/Contents/lib, version=243.21565.193, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.log.impl, descriptorPath=intellij.platform.vcs.log.impl.xml, path=~/Applications/IntelliJ IDEA Ultimate.app/Contents/lib, version=243.21565.193, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.dvcs.impl, descriptorPath=intellij.platform.vcs.dvcs.impl.xml, path=~/Applications/IntelliJ IDEA Ultimate.app/Contents/lib, version=243.21565.193, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.collaborationTools, descriptorPath=intellij.platform.collaborationTools.xml, path=~/Applications/IntelliJ IDEA Ultimate.app/Contents/lib, version=243.21565.193, package=null, isBundled=true), )
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.kt:157)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 105 more
```

**Comment by chengpohi** [comment]
Got it, I will fix this ASAP, Thanks your feedback

**Comment by chengpohi** [comment]
Hi @vvandens , this incompatiblity issue is fixed by 1.0.16, you can download from Jetbrains market [v1.10.16](https://plugins.jetbrains.com/plugin/16364-elasticsearch-query--edql/edit/versions/stable/635617) or manual install by [Github release v1.10.16](https://github.com/chengpohi/edql/releases/tag/v1.10.16).

Thanks your feedback 👍👍👍

**Comment by vvandens** [comment]
Thank you @chengpohi for the very quick fix !
New version not detected yet from my IntelliJ installation but it should arrive soon ;-)
